File: ExcelReports_VB/ReportEngine.vb
Public Class ReportEngine

    Public Shared Sub CreateExcelDocument(ByVal TemplateFileName As String, ByVal Parameters As List(Of ER_Common.Str_Parameter), ByRef Ds_Source As DataSet, ByRef Ds_Source_Pivot As DataSet, ByRef Ds_Source_Pivot_Desc As DataSet, ByRef Ds_Source_Pivot_Totals As DataSet, ByRef SaveFileName As String, Optional ByRef IsProtected As Boolean = False, Optional ByRef FileFormat As ER_Common.eExcelFileFormat = ER_Common.eExcelFileFormat.xlNormal)
        Dim Sp As New ClsParameters()
        Parameters.ForEach(Sub(O) Sp.Add(O.Name, O.Value))

        Dim NxlFileFormat As NativeExcel.XlFileFormat = Methods.ParseEnum(Of NativeExcel.XlFileFormat)(FileFormat.ToString())

        Methods_NativeExcel.NativeExcel_CreateExcelDocument(TemplateFileName, Sp, Ds_Source, Ds_Source_Pivot, Ds_Source_Pivot_Desc, Ds_Source_Pivot_Totals, SaveFileName, IsProtected, NxlFileFormat)
    End Sub

End Class

File: ExcelReports/ReportEngine.cs
using System;
using System.Collections.Generic;
//using System.Linq;
using System.Text;
using System.Data;

namespace ExcelReports
{
    public class ReportEngine
    {
        public static void CreateExcelDocument(
        String TemplateFileName
        , List<ER_Common.Str_Parameter> Parameters
        , DataSet Ds_Source
        , DataSet Ds_Source_Pivot
        , DataSet Ds_Source_Pivot_Desc
        , DataSet Ds_Source_Pivot_Totals
        , String SaveFileName
        , Boolean IsProtected = false
        , ER_Common.eExcelFileFormat FileFormat = ER_Common.eExcelFileFormat.xlNormal)
        {
            ReportEngine_NativeExcel.CreateExcelDocument(
                TemplateFileName
                , Parameters
                , Ds_Source
                , Ds_Source_Pivot
                , Ds_Source_Pivot_Desc
                , Ds_Source_Pivot_Totals
                , SaveFileName
                , IsProtected
                , FileFormat);

            //ExcelReports_VB.ClsParameters P = new ExcelReports_VB.ClsParameters();
            //Parameters.ForEach(O => { P.Add(O.Name, O.Value); });

            //NativeExcel.XlFileFormat NxlFileFormat = ER_Common.ParseEnum<NativeExcel.XlFileFormat>(FileFormat.ToString());

            //ExcelReports_VB.Methods_NativeExcel.NativeExcel_CreateExcelDocument(
            //    TemplateFileName
            //    , P
            //    , Ds_Source
            //    , Ds_Source_Pivot
            //    , Ds_Source_Pivot_Desc
            //    , Ds_Source_Pivot_Totals
            //    , SaveFileName
            //    , IsProtected
            //    , NxlFileFormat);
        }

    }
}
